Some authorization adapter test failing on Java 17

### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

testsuite

### Describe the bug

For example running testsuite with this command on Java 17:

```
mvn -B -f testsuite/integration-arquillian/pom.xml clean install -Pauth-server-quarkus -Dtest=DefaultAuthzConfigAdapterTest
```
the test fails. The reason is, that class `java.lang.System` is not available in the JSP files, which are used by authorization adapter tests. This is failing when `app-server-undertow` is used for adapters testing (which is the default).

There are multiple authorization adapter tests affected by this issue. The whole stacktrace:

```
18:01:37,757 INFO  [org.keycloak.testsuite.arquillian.AuthServerTestEnricher] ************************app-server-undertow*****************************************************************************
18:01:37,833 INFO  [io.undertow] starting server: Undertow - 2.2.19.Final
18:01:37,837 INFO  [org.keycloak.testsuite.arquillian.undertow.UndertowAppServer] App server started in 80ms on http://0.0.0.0:8280/
18:01:38,260 INFO  [org.wildfly.security] ELY00001: WildFly Elytron version 1.19.0.Final
18:01:38,685 INFO  [org.keycloak.testsuite.authz.adapter.example.DefaultAuthzConfigAdapterTest] [DefaultAuthzConfigAdapterTest] testDefaultAuthzConfig() STARTED
Jan 02, 2023 6:01:39 PM org.jboss.arquillian.drone.webdriver.factory.HtmlUnitDriverFactory setClientOptions
INFO: Setting HtmlDriver web client options: {htmlunitCssEnabled=false, htmlunitHistoryPageCacheLimit=1}
18:01:39,664 INFO  [org.keycloak.testsuite.drone.KeycloakDronePostSetup] Configuring driver settings. implicitWait=1500, pageLoadTimeout=10000
18:01:40,309 DEBUG [org.keycloak.testsuite.authz.adapter.example.DefaultAuthzConfigAdapterTest] updating admin password
18:01:40,310 INFO  [org.keycloak.testsuite.util.URLUtils] starting navigation
18:01:40,311 INFO  [org.keycloak.testsuite.util.URLUtils] current URL:  about:blank
18:01:40,311 INFO  [org.keycloak.testsuite.util.URLUtils] navigating to https://localhost:8543/auth
18:01:40,608 INFO  [org.keycloak.testsuite.util.URLUtils] new current URL:  https://localhost:8543/auth/
18:01:40,609 INFO  [org.keycloak.testsuite.util.URLUtils] navigation complete
18:01:40,611 DEBUG [org.keycloak.testsuite.authz.adapter.example.DefaultAuthzConfigAdapterTest] loading test realms
Loaded realm hello-world-authz
18:01:40,711 INFO  [org.keycloak.testsuite.authz.adapter.example.DefaultAuthzConfigAdapterTest] Setting redirect-uris in test realm 'hello-world-authz' as non-relative
18:01:40,713 INFO  [org.keycloak.testsuite.authz.adapter.example.DefaultAuthzConfigAdapterTest] importing test realms
18:01:40,713 DEBUG [org.keycloak.testsuite.authz.adapter.example.DefaultAuthzConfigAdapterTest] --importing realm: hello-world-authz
18:01:42,589 INFO  [org.keycloak.testsuite.util.URLUtils] starting navigation
18:01:42,590 INFO  [org.keycloak.testsuite.util.URLUtils] current URL:  https://localhost:8543/auth/
18:01:42,590 INFO  [org.keycloak.testsuite.util.URLUtils] navigating to https://localhost:8543/auth/realms/test/super-random-page
18:01:42,743 INFO  [org.keycloak.testsuite.util.URLUtils] new current URL:  https://localhost:8543/auth/realms/test/super-random-page
18:01:42,743 INFO  [org.keycloak.testsuite.util.URLUtils] navigation complete
18:01:42,743 INFO  [org.keycloak.testsuite.authz.adapter.example.DefaultAuthzConfigAdapterTest] deleting cookies in 'test' realm
18:01:43,574 INFO  [org.keycloak.testsuite.arquillian.undertow.UndertowAppServer] Deploying archive hello-world-authz-service.war
18:01:43,782 WARN  [org.keycloak.testsuite.utils.undertow.UndertowDeployerHelper] Application 'hello-world-authz-service.war' did not found resource on path /index.html
2023-01-02 18:01:44,773 INFO  [org.keycloak.services] (executor-thread-0) KC-SERVICES0106: Created script engine 'OpenJDK Nashorn', version '15.3' for the mime type 'text/javascript'
18:01:44,998 WARN  [org.keycloak.testsuite.utils.undertow.UndertowDeployerHelper] Application 'hello-world-authz-service.war' did not found resource on path /WEB-INF/tagPlugins.xml
18:01:45,507 ERROR [org.apache.jasper.compiler] JBWEB005028: Error reading class file java.lang.System
org.eclipse.jdt.internal.compiler.classfmt.ClassFormatException
	at org.eclipse.jdt.internal.compiler.classfmt.ClassFileReader.<init>(ClassFileReader.java:406)
	at org.apache.jasper.compiler.JDTCompiler$1.findType(JDTCompiler.java:210)
	at org.apache.jasper.compiler.JDTCompiler$1.findType(JDTCompiler.java:182)
	at org.eclipse.jdt.internal.compiler.lookup.LookupEnvironment.askForType(LookupEnvironment.java:174)
	at org.eclipse.jdt.internal.compiler.lookup.PackageBinding.getType(PackageBinding.java:145)
	at org.eclipse.jdt.internal.compiler.lookup.Scope.findType(Scope.java:1926)
	at org.eclipse.jdt.internal.compiler.lookup.Scope.getTypeOrPackage(Scope.java:3254)
	at org.eclipse.jdt.internal.compiler.lookup.Scope.getBinding(Scope.java:2192)
	at org.eclipse.jdt.internal.compiler.ast.SingleNameReference.resolveType(SingleNameReference.java:987)
	at org.eclipse.jdt.internal.compiler.ast.MessageSend.resolveType(MessageSend.java:636)
	at org.eclipse.jdt.internal.compiler.ast.MessageSend.resolveType(MessageSend.java:684)
	at org.eclipse.jdt.internal.compiler.ast.LocalDeclaration.resolve(LocalDeclaration.java:252)
	at org.eclipse.jdt.internal.compiler.ast.Block.resolveUsing(Block.java:130)
	at org.eclipse.jdt.internal.compiler.ast.TryStatement.resolve(TryStatement.java:1073)
	at org.eclipse.jdt.internal.compiler.ast.AbstractMethodDeclaration.resolveStatements(AbstractMethodDeclaration.java:634)
	at org.eclipse.jdt.internal.compiler.ast.MethodDeclaration.resolveStatements(MethodDeclaration.java:306)
	at org.eclipse.jdt.internal.compiler.ast.AbstractMethodDeclaration.resolve(AbstractMethodDeclaration.java:544)
	at org.eclipse.jdt.internal.compiler.ast.TypeDeclaration.resolve(TypeDeclaration.java:1195)
	at org.eclipse.jdt.internal.compiler.ast.TypeDeclaration.resolve(TypeDeclaration.java:1308)
	at org.eclipse.jdt.internal.compiler.ast.CompilationUnitDeclaration.resolve(CompilationUnitDeclaration.java:593)
	at org.eclipse.jdt.internal.compiler.Compiler.process(Compiler.java:867)
	at org.eclipse.jdt.internal.compiler.Compiler.processCompiledUnits(Compiler.java:550)
	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:462)
	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:417)
	at org.apache.jasper.compiler.JDTCompiler.generateClass(JDTCompiler.java:452)
	at org.apache.jasper.compiler.Compiler.compile(Compiler.java:375)
	at org.apache.jasper.compiler.Compiler.compile(Compiler.java:347)
	at org.apache.jasper.compiler.Compiler.compile(Compiler.java:331)
	at org.apache.jasper.JspCompilationContext.compile(JspCompilationContext.java:667)
	at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:363)
	at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:403)
	at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:347)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:590)
	at io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:74)
	at io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:62)
	at io.undertow.servlet.handlers.ServletChain$1.handleRequest(ServletChain.java:68)
	at io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36)
	at org.keycloak.adapters.undertow.UndertowAuthenticatedActionsHandler.handleRequest(UndertowAuthenticatedActionsHandler.java:66)
	at io.undertow.servlet.handlers.RedirectDirHandler.handleRequest(RedirectDirHandler.java:68)
	at io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:117)
	at io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:57)
	at io.undertow.server.handlers.DisableCacheHandler.handleRequest(DisableCacheHandler.java:33)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.security.handlers.AuthenticationConstraintHandler.handleRequest(AuthenticationConstraintHandler.java:53)
	at io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:46)
	at io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:64)
	at io.undertow.servlet.handlers.security.ServletSecurityConstraintHandler.handleRequest(ServletSecurityConstraintHandler.java:59)
	at io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:60)
	at io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:77)
	at io.undertow.security.handlers.AbstractSecurityContextAssociationHandler.handleRequest(AbstractSecurityContextAssociationHandler.java:43)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at org.keycloak.adapters.undertow.ServletPreAuthActionsHandler.handleRequest(ServletPreAuthActionsHandler.java:69)
	at io.undertow.servlet.handlers.SendErrorPageHandler.handleRequest(SendErrorPageHandler.java:52)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:275)
	at io.undertow.servlet.handlers.ServletInitialHandler.access$100(ServletInitialHandler.java:79)
	at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:134)
	at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:131)
	at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:48)
	at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
	at io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:255)
	at io.undertow.servlet.handlers.ServletInitialHandler.access$000(ServletInitialHandler.java:79)
	at io.undertow.servlet.handlers.ServletInitialHandler$1.handleRequest(ServletInitialHandler.java:100)
	at io.undertow.server.Connectors.executeRootHandler(Connectors.java:387)
	at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:852)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1982)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1486)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1377)
	at org.xnio.XnioWorker$WorkerThreadFactory$1$1.run(XnioWorker.java:1282)
	at java.base/java.lang.Thread.run(Thread.java:833)
18:01:45,556 ERROR [io.undertow.request] UT005023: Exception handling request to /hello-world-authz-service/
org.apache.jasper.JasperException: JBWEB004062: Unable to compile class for JSP: 

JBWEB004060: An error occurred at line: 28 in the jsp file: /index.jsp
System cannot be resolved
25:     KeycloakSecurityContext keycloakSecurityContext = (KeycloakSecurityContext) request.getAttribute(KeycloakSecurityContext.class.getName());
26:     AuthorizationContext authzContext = keycloakSecurityContext.getAuthorizationContext();
27: 
28:     boolean isTLSEnabled = Boolean.parseBoolean(System.getProperty("auth.server.ssl.required", "true"));
29:     String authPort = isTLSEnabled ? System.getProperty("auth.server.https.port", "8543") : System.getProperty("auth.server.http.port", "8180");
30:     String authScheme = isTLSEnabled ? "https" : "http";
31:     String authUri = authScheme + "://localhost:" + authPort + "/auth";


JBWEB004060: An error occurred at line: 29 in the jsp file: /index.jsp
System cannot be resolved
26:     AuthorizationContext authzContext = keycloakSecurityContext.getAuthorizationContext();
27: 
28:     boolean isTLSEnabled = Boolean.parseBoolean(System.getProperty("auth.server.ssl.required", "true"));
29:     String authPort = isTLSEnabled ? System.getProperty("auth.server.https.port", "8543") : System.getProperty("auth.server.http.port", "8180");
30:     String authScheme = isTLSEnabled ? "https" : "http";
31:     String authUri = authScheme + "://localhost:" + authPort + "/auth";
32: %>


JBWEB004060: An error occurred at line: 29 in the jsp file: /index.jsp
System cannot be resolved
26:     AuthorizationContext authzContext = keycloakSecurityContext.getAuthorizationContext();
27: 
28:     boolean isTLSEnabled = Boolean.parseBoolean(System.getProperty("auth.server.ssl.required", "true"));
29:     String authPort = isTLSEnabled ? System.getProperty("auth.server.https.port", "8543") : System.getProperty("auth.server.http.port", "8180");
30:     String authScheme = isTLSEnabled ? "https" : "http";
31:     String authUri = authScheme + "://localhost:" + authPort + "/auth";
32: %>


Stacktrace:
	at org.apache.jasper.compiler.DefaultErrorHandler.javacError(DefaultErrorHandler.java:95)
	at org.apache.jasper.compiler.ErrorDispatcher.javacError(ErrorDispatcher.java:212)
	at org.apache.jasper.compiler.JDTCompiler.generateClass(JDTCompiler.java:462)
	at org.apache.jasper.compiler.Compiler.compile(Compiler.java:375)
	at org.apache.jasper.compiler.Compiler.compile(Compiler.java:347)
	at org.apache.jasper.compiler.Compiler.compile(Compiler.java:331)
	at org.apache.jasper.JspCompilationContext.compile(JspCompilationContext.java:667)
	at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:363)
	at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:403)
	at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:347)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:590)
	at io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:74)
	at io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:62)
	at io.undertow.servlet.handlers.ServletChain$1.handleRequest(ServletChain.java:68)
	at io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36)
	at org.keycloak.adapters.undertow.UndertowAuthenticatedActionsHandler.handleRequest(UndertowAuthenticatedActionsHandler.java:66)
	at io.undertow.servlet.handlers.RedirectDirHandler.handleRequest(RedirectDirHandler.java:68)
	at io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:117)
	at io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:57)
	at io.undertow.server.handlers.DisableCacheHandler.handleRequest(DisableCacheHandler.java:33)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.security.handlers.AuthenticationConstraintHandler.handleRequest(AuthenticationConstraintHandler.java:53)
	at io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:46)
	at io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:64)
	at io.undertow.servlet.handlers.security.ServletSecurityConstraintHandler.handleRequest(ServletSecurityConstraintHandler.java:59)
	at io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:60)
	at io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:77)
	at io.undertow.security.handlers.AbstractSecurityContextAssociationHandler.handleRequest(AbstractSecurityContextAssociationHandler.java:43)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at org.keycloak.adapters.undertow.ServletPreAuthActionsHandler.handleRequest(ServletPreAuthActionsHandler.java:69)
	at io.undertow.servlet.handlers.SendErrorPageHandler.handleRequest(SendErrorPageHandler.java:52)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:275)
	at io.undertow.servlet.handlers.ServletInitialHandler.access$100(ServletInitialHandler.java:79)
	at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:134)
	at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:131)
	at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:48)
	at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
	at io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:255)
	at io.undertow.servlet.handlers.ServletInitialHandler.access$000(ServletInitialHandler.java:79)
	at io.undertow.servlet.handlers.ServletInitialHandler$1.handleRequest(ServletInitialHandler.java:100)
	at io.undertow.server.Connectors.executeRootHandler(Connectors.java:387)
	at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:852)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1982)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1486)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1377)
	at org.xnio.XnioWorker$WorkerThreadFactory$1$1.run(XnioWorker.java:1282)
	at java.base/java.lang.Thread.run(Thread.java:833)
18:01:45,587 INFO  [org.keycloak.testsuite.arquillian.undertow.UndertowAppServer] Undeploying archive hello-world-authz-service.war
18:01:45,594 INFO  [org.keycloak.testsuite.authz.adapter.example.DefaultAuthzConfigAdapterTest] removing test realms after test method
18:01:45,594 INFO  [org.keycloak.testsuite.authz.adapter.example.DefaultAuthzConfigAdapterTest] removing realm: hello-world-authz
18:01:46,297 ERROR [org.keycloak.testsuite.authz.adapter.example.DefaultAuthzConfigAdapterTest] [DefaultAuthzConfigAdapterTest] testDefaultAuthzConfig() FAILED
18:01:46,298 INFO  [org.keycloak.testsuite.authz.adapter.example.DefaultAuthzConfigAdapterTest] [DefaultAuthzConfigAdapterTest] testDefaultAuthzConfig() FINISHED


18:01:46,299 INFO  [org.keycloak.testsuite.arquillian.AppServerTestEnricher] Stopping app server: app-server-undertow
Jan 02, 2023 6:01:46 PM org.jboss.arquillian.container.test.impl.client.container.ClientContainerController stop
INFO: Manual stopping of a server instance
18:01:46,300 INFO  [io.undertow] stopping server: Undertow - 2.2.19.Final
18:01:46,327 INFO  [org.keycloak.testsuite.arquillian.undertow.UndertowAppServer] App Server stopped.
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 23.094 s <<< FAILURE! - in org.keycloak.testsuite.authz.adapter.example.DefaultAuthzConfigAdapterTest
[ERROR] org.keycloak.testsuite.authz.adapter.example.DefaultAuthzConfigAdapterTest.testDefaultAuthzConfig  Time elapsed: 7.611 s  <<< FAILURE!
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:87)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at org.junit.Assert.assertTrue(Assert.java:53)
	at org.keycloak.testsuite.authz.adapter.example.DefaultAuthzConfigAdapterTest.testDefaultAuthzConfig(DefaultAuthzConfigAdapterTest.java:90)

```

### Version

20.0.2

--- a/testsuite/integration-arquillian/test-apps/hello-world-authz-service/src/main/java/org/keycloak/testsuite/example/WebAppHelper.java
+++ b/testsuite/integration-arquillian/test-apps/hello-world-authz-service/src/main/java/org/keycloak/testsuite/example/WebAppHelper.java
@@ -1,0 +1,34 @@
+/*
+ * Copyright 2023 Red Hat, Inc. and/or its affiliates
+ *  and other contributors as indicated by the @author tags.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package org.keycloak.testsuite.example;
+
+/**
+ * @author <a href="mailto:mposolda@redhat.com">Marek Posolda</a>
+ */
+public class WebAppHelper {
+
+    public static String getAuthUri() {
+        boolean isTLSEnabled = Boolean.parseBoolean(System.getProperty("auth.server.ssl.required", "true"));
+        String authPort = isTLSEnabled ? System.getProperty("auth.server.https.port", "8543") : System.getProperty("auth.server.http.port", "8180");
+        String authScheme = isTLSEnabled ? "https" : "http";
+        return authScheme + "://localhost:" + authPort + "/auth";
+    }
+
+}

--- a/testsuite/integration-arquillian/test-apps/hello-world-authz-service/src/main/webapp/error.jsp
+++ b/testsuite/integration-arquillian/test-apps/hello-world-authz-service/src/main/webapp/error.jsp
@@ -18,17 +18,11 @@
 
 <%@ page import="org.keycloak.common.util.KeycloakUriBuilder" %>
 <%@ page import="org.keycloak.constants.ServiceUrlConstants" %>
-
-<%
-    boolean isTLSEnabled = Boolean.parseBoolean(System.getProperty("auth.server.ssl.required", "true"));
-    String authPort = isTLSEnabled ? System.getProperty("auth.server.https.port", "8543") : System.getProperty("auth.server.http.port", "8180");
-    String authScheme = isTLSEnabled ? "https" : "http";
-    String authUri = authScheme + "://localhost:" + authPort + "/auth";
-%>
+<%@ page import="org.keycloak.testsuite.example.WebAppHelper" %>
 
 <html>
 <body>
-<h2><a href="<%= KeycloakUriBuilder.fromUri(authUri).path(ServiceUrlConstants.TOKEN_SERVICE_LOGOUT_PATH)
+<h2><a href="<%= KeycloakUriBuilder.fromUri(WebAppHelper.getAuthUri()).path(ServiceUrlConstants.TOKEN_SERVICE_LOGOUT_PATH)
             .queryParam("redirect_uri", "http://localhost:8080/hello-world-authz-service").build("hello-world-authz").toString()%>">Logout</a></h2>
 
 <h3>Access Denied !</h3>

--- a/testsuite/integration-arquillian/test-apps/hello-world-authz-service/src/main/webapp/index.jsp
+++ b/testsuite/integration-arquillian/test-apps/hello-world-authz-service/src/main/webapp/index.jsp
@@ -20,20 +20,16 @@
 <%@ page import="org.keycloak.common.util.KeycloakUriBuilder" %>
 <%@ page import="org.keycloak.constants.ServiceUrlConstants" %>
 <%@ page import="org.keycloak.representations.idm.authorization.Permission" %>
+<%@ page import="org.keycloak.testsuite.example.WebAppHelper" %>
 
 <%
     KeycloakSecurityContext keycloakSecurityContext = (KeycloakSecurityContext) request.getAttribute(KeycloakSecurityContext.class.getName());
     AuthorizationContext authzContext = keycloakSecurityContext.getAuthorizationContext();
-
-    boolean isTLSEnabled = Boolean.parseBoolean(System.getProperty("auth.server.ssl.required", "true"));
-    String authPort = isTLSEnabled ? System.getProperty("auth.server.https.port", "8543") : System.getProperty("auth.server.http.port", "8180");
-    String authScheme = isTLSEnabled ? "https" : "http";
-    String authUri = authScheme + "://localhost:" + authPort + "/auth";
 %>
 <html>
 <body>
 <h2>Welcome !</h2>
-<h2><a href="<%= KeycloakUriBuilder.fromUri(authUri).path(ServiceUrlConstants.TOKEN_SERVICE_LOGOUT_PATH)
+<h2><a href="<%= KeycloakUriBuilder.fromUri(WebAppHelper.getAuthUri()).path(ServiceUrlConstants.TOKEN_SERVICE_LOGOUT_PATH)
             .queryParam("redirect_uri", "http://localhost:8080/hello-world-authz-service").build("hello-world-authz").toString()%>">Logout</a></h2>
 
 <h3>Your permissions are:</h3>

--- a/testsuite/integration-arquillian/test-apps/servlet-authz/src/main/java/org/keycloak/testsuite/servletauthz/WebAppHelper.java
+++ b/testsuite/integration-arquillian/test-apps/servlet-authz/src/main/java/org/keycloak/testsuite/servletauthz/WebAppHelper.java
@@ -1,0 +1,33 @@
+/*
+ * Copyright 2023 Red Hat, Inc. and/or its affiliates
+ *  and other contributors as indicated by the @author tags.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package org.keycloak.testsuite.servletauthz;
+
+/**
+ * @author <a href="mailto:mposolda@redhat.com">Marek Posolda</a>
+ */
+public class WebAppHelper {
+
+    public static String getAuthUri() {
+        boolean isTLSEnabled = Boolean.parseBoolean(System.getProperty("auth.server.ssl.required", "true"));
+        String authPort = isTLSEnabled ? System.getProperty("auth.server.https.port", "8543") : System.getProperty("auth.server.http.port", "8180");
+        String authScheme = isTLSEnabled ? "https" : "http";
+        return authScheme + "://localhost:" + authPort + "/auth";
+    }
+}

--- a/testsuite/integration-arquillian/test-apps/servlet-authz/src/main/webapp/logout-include.jsp
+++ b/testsuite/integration-arquillian/test-apps/servlet-authz/src/main/webapp/logout-include.jsp
@@ -1,18 +1,12 @@
 <%@ page import="org.keycloak.common.util.KeycloakUriBuilder" %>
 <%@ page import="org.keycloak.constants.ServiceUrlConstants" %>
+<%@ page import="org.keycloak.testsuite.servletauthz.WebAppHelper" %>
 <%
     String scheme = request.getScheme();
     String host = request.getServerName();
     int port = request.getServerPort();
     String contextPath = request.getContextPath();
     String redirectUri = scheme + "://" + host + ":" + port + contextPath;
-
-
-    boolean isTLSEnabled = Boolean.parseBoolean(System.getProperty("auth.server.ssl.required", "true"));
-    String authPort = isTLSEnabled ? System.getProperty("auth.server.https.port", "8543") : System.getProperty("auth.server.http.port", "8180");
-    String authScheme = isTLSEnabled ? "https" : "http";
-    String authHost = System.getProperty("auth.server.host", "localhost");
-    String authUri = authScheme + "://" + authHost + ":" + authPort + "/auth";
 %>
-<h2>Click here <a href="<%= KeycloakUriBuilder.fromUri(authUri).path(ServiceUrlConstants.TOKEN_SERVICE_LOGOUT_PATH)
+<h2>Click here <a href="<%= KeycloakUriBuilder.fromUri(WebAppHelper.getAuthUri()).path(ServiceUrlConstants.TOKEN_SERVICE_LOGOUT_PATH)
             .build("servlet-authz").toString()%>">Sign Out</a></h2>

--- a/testsuite/integration-arquillian/test-apps/servlet-policy-enforcer/src/main/java/org/keycloak/testsuite/example/WebAppHelper.java
+++ b/testsuite/integration-arquillian/test-apps/servlet-policy-enforcer/src/main/java/org/keycloak/testsuite/example/WebAppHelper.java
@@ -1,0 +1,33 @@
+/*
+ * Copyright 2023 Red Hat, Inc. and/or its affiliates
+ *  and other contributors as indicated by the @author tags.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package org.keycloak.testsuite.example;
+
+/**
+ * @author <a href="mailto:mposolda@redhat.com">Marek Posolda</a>
+ */
+public class WebAppHelper {
+
+    public static String getAuthUri() {
+        boolean isTLSEnabled = Boolean.parseBoolean(System.getProperty("auth.server.ssl.required", "true"));
+        String authPort = isTLSEnabled ? System.getProperty("auth.server.https.port", "8543") : System.getProperty("auth.server.http.port", "8180");
+        String authScheme = isTLSEnabled ? "https" : "http";
+        return authScheme + "://localhost:" + authPort + "/auth";
+    }
+}

--- a/testsuite/integration-arquillian/test-apps/servlet-policy-enforcer/src/main/webapp/logout-include.jsp
+++ b/testsuite/integration-arquillian/test-apps/servlet-policy-enforcer/src/main/webapp/logout-include.jsp
@@ -1,18 +1,12 @@
 <%@ page import="org.keycloak.common.util.KeycloakUriBuilder" %>
 <%@ page import="org.keycloak.constants.ServiceUrlConstants" %>
+<%@ page import="org.keycloak.testsuite.example.WebAppHelper" %>
 <%
     String scheme = request.getScheme();
     String host = request.getServerName();
     int port = request.getServerPort();
     String contextPath = request.getContextPath();
     String redirectUri = scheme + "://" + host + ":" + port + contextPath;
-
-
-    boolean isTLSEnabled = Boolean.parseBoolean(System.getProperty("auth.server.ssl.required", "true"));
-    String authPort = isTLSEnabled ? System.getProperty("auth.server.https.port", "8543") : System.getProperty("auth.server.http.port", "8180");
-    String authScheme = isTLSEnabled ? "https" : "http";
-    String authHost = System.getProperty("auth.server.host", "localhost");
-    String authUri = authScheme + "://" + authHost + ":" + authPort + "/auth";
 %>
-<h2>Click here <a href="<%= KeycloakUriBuilder.fromUri(authUri).path(ServiceUrlConstants.TOKEN_SERVICE_LOGOUT_PATH)
+<h2>Click here <a href="<%= KeycloakUriBuilder.fromUri(WebAppHelper.getAuthUri()).path(ServiceUrlConstants.TOKEN_SERVICE_LOGOUT_PATH)
             .build("servlet-policy-enforcer-authz").toString()%>">Sign Out</a></h2>

